src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs

```diff
@@ -2,31 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Collections;
 using System.ComponentModel;
 
 namespace System.Windows.Forms;
 
 internal class RelatedPropertyManager : PropertyManager
 {
-    private BindingManagerBase parentManager;
-    private string dataField;
-    private PropertyDescriptor fieldInfo;
+    private BindingManagerBase _parentManager;
+    private PropertyDescriptor _fieldInfo;
 
-    internal RelatedPropertyManager(BindingManagerBase parentManager, string dataField) : base(GetCurrentOrNull(parentManager), dataField)
+    internal RelatedPropertyManager(BindingManagerBase parentManager, string dataField)
+        : base(GetCurrentOrNull(parentManager), dataField)
     {
         Bind(parentManager, dataField);
     }
 
+    [MemberNotNull(nameof(_parentManager))]
+    [MemberNotNull(nameof(_fieldInfo))]
     private void Bind(BindingManagerBase parentManager, string dataField)
     {
         Debug.Assert(parentManager is not null, "How could this be a null parentManager.");
-        this.parentManager = parentManager;
-        this.dataField = dataField;
-        fieldInfo = parentManager.GetItemProperties().Find(dataField, true);
-        if (fieldInfo is null)
+        _parentManager = parentManager;
+        _fieldInfo = parentManager.GetItemProperties().Find(dataField, true)!;
+        if (_fieldInfo is null)
```

Comment:
May be changing to throw right away looks cleaner.
```suggestion
        _fieldInfo = parentManager.GetItemProperties().Find(dataField, true) ?? throw new ArgumentException(string.Format(SR.RelatedListManagerChild, dataField));
```